tests/components/ajax/aggregate.test.js

```diff
@@ -0,0 +1,245 @@
+
+      ajaxAggregate.ajaxEvents = [] // clear ajaxEvents buffer
+    })
+
+    test('for a (old) SPA ajax request buffers in spaAjaxEvents', () => {
+      const interaction = { id: 0 }
+      context.spaNode = { interaction }
+
+      ajaxAggregate.ee.emit('xhr', ajaxArguments, context)
+
+      const interactionAjaxEvents = ajaxAggregate.spaAjaxEvents[interaction.id]
+      expect(interactionAjaxEvents.length).toEqual(1) // SPA ajax requests are buffered in spaAjaxEvents and under its interaction id
+      expect(ajaxAggregate.ajaxEvents.length).toEqual(0)
+
+      const spaAjaxEvent = interactionAjaxEvents[0]
+      expect(spaAjaxEvent).toEqual(expect.objectContaining({ startTime: 0, path: '/pathname' }))
+
+      ajaxAggregate.spaAjaxEvents = {} // clear spaAjaxEvents
+    })
+
+    test('for ajax under soft nav does not buffer and instead pipes it', () => {
+      softNavInUse = true
+
+      ajaxAggregate.ee.emit('xhr', ajaxArguments, context)
+
+      expect(ajaxAggregate.ajaxEvents.length).toEqual(0)
+      expect(Object.keys(ajaxAggregate.spaAjaxEvents).length).toEqual(0)
+      expect(hMod.handle).toHaveBeenLastCalledWith('ajax', [expect.objectContaining({ startTime: 0, path: '/pathname' })],
+        undefined, FEATURE_NAMES.softNav, expect.any(Object))
+
+      softNavInUse = false
+    })
+  })
+
+  test('on interactionDiscarded, saved (old) SPA events are put back in ajaxEvents', () => {
+    const interaction = { id: 0 }
+    context.spaNode = { interaction }
+
+    ajaxAggregate.ee.emit('xhr', ajaxArguments, context)
+    ajaxAggregate.ee.emit('interactionDone', [interaction, false])
+
+    expect(ajaxAggregate.spaAjaxEvents[interaction.id]).toBeUndefined() // no interactions in SPA under interaction 0
+    expect(ajaxAggregate.ajaxEvents.length).toEqual(1)
+
+    ajaxAggregate.ajaxEvents = []
+  })
+  test('on returnAjax from soft nav, event is re-routed back into ajaxEvents', () => {
+    softNavInUse = true
+
+    ajaxAggregate.ee.emit('xhr', ajaxArguments, context)
+    const event = hMod.handle.mock.lastCall[1][0]
+    ajaxAggregate.ee.emit('returnAjax', [event], context)
+
+    expect(ajaxAggregate.ajaxEvents.length).toEqual(1)
+    expect(ajaxAggregate.ajaxEvents[0]).toEqual(expect.objectContaining({ startTime: 0, path: '/pathname' }))
+
+    softNavInUse = false
+  })
+
+  describe('prepareHarvest', () => {
+    test('correctly serializes an AjaxRequest events payload', () => {
+      const expected = {
+        type: 'ajax',
+        start: 0,
+        end: 30,
+        callbackEnd: 30,
+        callbackDuration: 0,
+        domain: 'https://example.com',
+        path: '/pathname',
+        method: 'PUT',
+        status: 200,
+        requestedWith: 'XMLHttpRequest',
+        requestBodySize: 128,
+        responseBodySize: 256,
+        nodeId: '0',
+        guid: null,
+        traceId: null,
+        timestamp: null
+      }
+
+      ajaxAggregate.ee.emit('xhr', ajaxArguments, context)
```

Comment:
The [original test had two ajax's](https://github.com/newrelic/newrelic-browser-agent/blob/c487e04f46a606545c8172b6ed1c022b4dc1cac1/tests/browser/xhr/ajax-events.browser.js#L176-L177). Theres not a test in here that two ajaxs that arent "too big" are harvested correctly.  Should consider if that is a test case that needs evaluation

Reply:
I took that as a typo. The test right underneath it `correctly serializes a very large AjaxRequest events payload` queues up 10 ajax events.